test/core/test/spy.test.ts

```diff
@@ -1,5 +1,7 @@
 import { describe, expect, test, vi } from 'vitest'
 import * as mock from './fixtures/hello-mock'
+import * as squaredModule from './fixtures/squared'
```

Comment:
We need more tests for this behaviour. I also added two more issues in your description - you can look it up there. The tests should be in examples/mocks